power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -739,49 +757,102 @@ class Deserializer {
             throw SerializationError{
                 "An element of a list should have same length as the list of predefined attributes!\n"};
         }
+
         for (attribute_number_ = 0; attribute_number_ != array_size; ++attribute_number_) {
-            parse_attribute(element_pointer, *attributes[attribute_number_]);
+            parse_attribute(buffer, component, *attributes[attribute_number_]);
         }
         attribute_number_ = -1;
     }
 
-    void parse_attribute(void* element_pointer, MetaAttribute const& attribute) {
+    void parse_attribute(BufferView const& buffer, MetaComponent const& component, MetaAttribute const& attribute) {
+        assert(buffer.buffer != nullptr);
+
+        if (buffer.buffer->data != nullptr) {
+            parse_attribute(row_based, buffer, component, attribute);
+        } else {
+            parse_attribute(columnar, buffer, component, attribute);
+        }
+    }
+
+    void parse_attribute(row_based_t /*tag*/, BufferView const& buffer, MetaComponent const& component,
+                         MetaAttribute const& attribute) {
+        // call relevant parser
+        assert(buffer.buffer != nullptr);
+        assert(buffer.buffer->data != nullptr);
+
+        ctype_func_selector(attribute.ctype, [&buffer, &component, &attribute, this]<class T> {
+            ValueVisitor<T> visitor{{},
+                                    attribute.get_attribute<T>(component.advance_ptr(buffer.buffer->data, buffer.idx))};
+            msgpack::parse(data_, size_, offset_, visitor);
+        });
+    }
+
+    void parse_attribute(columnar_t /*tag*/, BufferView const& buffer, MetaComponent const& /*component*/,
+                         MetaAttribute const& attribute) {
         // call relevant parser
-        ctype_func_selector(attribute.ctype, [element_pointer, &attribute, this]<class T> {
-            ValueVisitor<T> visitor{{}, attribute.get_attribute<T>(element_pointer)};
+        assert(buffer.buffer != nullptr);
+        assert(buffer.buffer->data == nullptr);
+
+        if (attribute_number_ >= 0 && buffer.attribute_order != nullptr) {
+            assert(attribute_number_ < std::ssize(*buffer.attribute_order));
+
+            if (auto const* const attribute_buffer = (*buffer.attribute_order)[attribute_number_];
+                attribute_buffer != nullptr) {
+                parse_attribute(*attribute_buffer, buffer.idx);
+            } else {
+                parse_skip();
+            }
+        } else if (auto it = std::ranges::find_if(buffer.buffer->attributes,
+                                                  [&attribute](auto const& attribute_buffer) {
+                                                      return attribute_buffer.meta_attribute == &attribute;
+                                                  });
+                   it != buffer.buffer->attributes.end()) {
+            parse_attribute(*it, buffer.idx);
+        } else {
+            parse_skip();
+        }
+    }
```

Comment:
I feel this logic as not necessary in this level. In this level, we should directly have a columnar buffer and an `MetaAttribute` reference. Nothing else.

In the `parse_map_element` and `parse_array_element`, the logic should already divide to resolve different path.